examples/with-claims/api/index.ts

```diff
@@ -0,0 +1,201 @@
+import EmailPassword from "supertokens-node/recipe/emailpassword";
+import { RolesClaim } from "./claims/RolesClaim";
+import { EmailVerifiedClaim } from "./claims/EmailVerifiedClaim";
+import { SecondFactorClaimValidators, SecondFactorOTPClaim } from "./claims/MFAClaim";
+import { roleDB } from "./mockRoleDb";
+
+// Change these values if you want to run the server on another adress
+const apiPort = process.env.REACT_APP_API_PORT || 3001;
+const apiDomain = process.env.REACT_APP_API_URL || `http://localhost:${apiPort}`;
+const websitePort = process.env.REACT_APP_WEBSITE_PORT || 3000;
+const websiteDomain = process.env.REACT_APP_WEBSITE_URL || `http://localhost:${websitePort}`;
+
+supertokens.init({
+    framework: "express",
+    supertokens: {
+        // TODO: This is a core hosted for demo purposes. You can use this, but make sure to change it to your core instance URI eventually.
+        connectionURI: "https://try.supertokens.io",
+    },
+    appInfo: {
+        appName: "SuperTokens Demo App", // TODO: Your app name
+        apiDomain, // TODO: Change to your app's API domain
+        websiteDomain, // TODO: Change to your app's website domain
+    },
+    recipeList: [
+        EmailPassword.init({
+            override: {
+                functions: (oI) => ({
+                    ...oI,
+                    signUp: async function (input) {
+                        const res = await oI.signUp(input);
+                        if (res.status === "OK") {
+                            if (res.user.email.includes("admin")) {
+                                roleDB.set(res.user.id, ["admin"]);
+                            } else {
+                                roleDB.set(res.user.id, ["user"]);
+                            }
+                        }
+                        return res;
+                    },
+                }),
+            },
+        }),
+        Session.init({
+            claimsToAddOnCreation: [
+                RolesClaim,
+                EmailVerifiedClaim,
+                ({ userId }) => ({ "app-custom-claim": { userId, why: "Info" } }),
+            ],
+            defaultValidatorsForVerification: [EmailVerifiedClaim.isVerified],
+        }),
+    ],
+});
+
+const app = express();
+app.use(
+    cors({
+        origin: websiteDomain, // TODO: Change to your app's website domain
+        allowedHeaders: ["content-type", ...supertokens.getAllCORSHeaders()],
+        methods: ["GET", "PUT", "POST", "DELETE"],
+        credentials: true,
+    })
+);
+
+app.use(
+    helmet({
+        contentSecurityPolicy: false,
+    })
+);
+app.use(morgan("dev") as any);
+app.use(bodyParser.json());
+
+app.use(middleware());
+
+app.get(
+    "/api1",
+    verifySession({
+        overwriteDefaultValidators: [
+            RolesClaim.hasRole.including("admin"),
+            EmailVerifiedClaim.isVerified, // Maybe this could be just EmailVerifiedClaim
+            SecondFactorClaimValidators.any2Factors,
```

Comment:
We could add something like "claim combinators", so functions like `and` and `or`, but I don't see a practical way of chaining with `&&`.
I think we could cover almost all use-cases by providing functions like `includesOneOf`, `includesAll` etc on each claim. I think this feature can launch without those combinators since it's solvable by overrides and/or custom claims. If this is a recurring thing we can implement those later.

Reply:
cool